### PR TITLE
Embed Page

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

```
<iframe src="http://localhost:3000/embed/level1-ZA_1_003/sectionIm8gSDP8/chartErJC8igE />
```

Screen Shot 2019-09-09 at 2 42 59 PM

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by karimkawambwa** [comment]
@kilemensi using unique ids here http://localhost:3000/embed/level1-ZA_1_003/sectionIm8gSDP8/chartErJC8igE

will also update https://github.com/TakwimuAfrica/Takwimu.AFRICA/pull/10

File: package.json [hunk 1]
```diff
@@ -46,12 +47,15 @@
     "prettier": "^1.16.1"
   },
   "scripts": {
+    "applyIdsToChartDefinitions": "node scripts/chartsDefinitionsApplyUniqueId.js",
```
Comment (kilemensi) [inline package.json]: `scrips` folder should also be added to `lint-staged`

File: package.json [hunk 2]
```diff
@@ -46,12 +47,15 @@
...
+    "prestart": "yarn applyIdsToChartDefinitions",
+    "prebuild": "yarn applyIdsToChartDefinitions",
+    "predeploy": "yarn run build",
```
Comment (kilemensi) [inline package.json]: Keeping `pre` and `post` commands around their respective action e.g.
```js
    "prestart": ....
    "start": ... 
    "poststart" ...
```
makes it easy to track what's happening me thinks.

File: package.json (hunk 1 above)
Comment (karimkawambwa) [reply package.json]: scrips ?

**Comment by karimkawambwa** [comment]
> 🚀 @karimkawambwa
> 
> 1. Lets extend the `value-id` URL scheme we're using for geography to sections and charts:
> 
> ```
>     profile/country-ZA
>     profile/level1-ZA_1_006
>     profile/country-ZA/section-SSSSSS/chart-CCCCC
>     profile/level1-ZA_1_006/section-SSSSSS/chart-CCCCC
> ```
> 
> 1. By doing insert ID on `deploy`, this means a separate commit (and PR) _might_ be necessary to push the updated `charts.json` to `master`. Can't we run the insert id command as part of `lint-staged`?

@kilemensi I have made the changes, please look at the package.json file again

**Comment by karimkawambwa** [comment]
> I'd have preferred if applyId is always followed by git add e.g. in the prestart & prebuild we're calling applyId but then we don't follow through, and

Only if we deploy should we commit since yarn start can be used just in development.

> postdeploy should push since we're commit-ing in predeploy.

I thought about doing, if you think it is worth then I'll add it.

cc @kilemensi

**Pull request merged**